Rewrote test suite documentation

krefson opened the pull request.

krefson committed:
Rewrote test suite documentation

oerc0122 commented on docs/developer/test_suite.md:
```diff
@@ -0,0 +1,465 @@
+
+### A complete test run
+
+A run of the test suite is started by running the testcode scipt `<CASTEP_ROOT>/bin/testcode.py` from within the Test didectory, but as some options are usually required, this is most conveniently invoked via the build system by either
```
```suggestion
A run of the test suite is started by running the testcode script `<CASTEP_ROOT>/bin/testcode.py` from within the Test directory, but as some options are usually required, this is most conveniently invoked via the build system by either
```

krefson replied:
OK

oerc0122 commented on docs/developer/test_suite.md:
```diff
@@ -0,0 +1,465 @@
+platform. This handles all of the definition of test categories, the
+selection, sequencing and running of tests, and comparison of the run
+output with a benchmark. To run CASTEP, postprocess the output and
+extract the comparison data it invokes two scripts supplied in
```
```suggestion
extract the comparison data, it invokes two scripts supplied in
```

krefson replied:
OK

oerc0122 commented on docs/developer/test_suite.md:
````diff
@@ -0,0 +1,465 @@
+............................................................................................................................................................................................................................................................................................................................................................................................................... ...
+```
+
+if everything passes or with `F` replacing one or more of the `.` symbols to indicate a failed test or tests.
````
Does the test-suite have skip (`S`) at all?

krefson replied:
Missed that, thanks.

oerc0122 commented on docs/developer/test_suite.md:
```diff
@@ -0,0 +1,465 @@
+or the targets of the same name in cmake.
+
+??? keyword "Running individual tests"
+    In fact the tests may be run at even finer granularity, down to the subdirectory. This is not implemented via the Makefile or cmake, and can only be accessed by invoking `testcode` directly.  The most straightforward means of doing this is to copy and paste the testcode invocation command printed by `make check-spe` including all of the run and argument setup and change the argument of the `- ...
```
> The most straightforward means of doing this is to copy and paste the testcode invocation command printed by `make check-spe`

Provide it here instead? While potentially liable to change I doubt the command will change sufficiently frequently to not warrant putting it here. 

At the very least make it clear the link to below contains the requisite lines.

krefson replied:
Reworded for clarity

oerc0122 commented on docs/developer/test_suite.md:
```diff
@@ -0,0 +1,465 @@
+  * If any external pseudopotential files are required, these should be placed in the `Test/Pseudopotentials` subdirectory and a symbolic link made from the test subdirectory. Both the fle and the link should be added to the `git` repository. 
+  * to modify [`bin/run_castep_test.py`](#test-outputs-and-benchmarks) if you need to test the contents of a new output file.
+  * if testing a new quantity, either in the `.castep` file or any other output file, you must extend the [`bin/extract_results.pl`](#extraction-of-properties) script to scan for and report the new value to be tested.
+  * A new Stanza in [`jobconfig`](#jobconfig-file) will be required to specify tolerances for the comparison if the defaults in [`userconfig`](#userconfig-file) are not suitable.
```
Does `stanza` need capitalising here?

krefson replied:
no

oerc0122 replied:
```suggestion
  * A new stanza in [`jobconfig`](#jobconfig-file) will be required to specify tolerances for the comparison if the defaults in [`userconfig`](#userconfig-file) are not suitable.
```

oerc0122 commented on docs/developer/test_suite.md:
```diff
@@ -0,0 +1,465 @@
+fail because no benchmark is present for comparison.
+
+    b. Copy or rename the just created test file
+"test.out.<date>?inp=<seedname>.param    to
+"benchmark.out.castep-<old-castep-version>castep-python-1.0.python-<old-python-version>.inp=<seedname>.param".
+where "<old-castep-version>" and "<old-python-version>" should be chosen to match the existing "benchmark.out" files from other tests.
```
```suggestion
"test.out.<date>?inp=<seedname>.param"  to
"benchmark.out.castep-<old-castep-version>castep-python-1.0.python-<old-python-version>.inp=<seedname>.param".
where "<old-castep-version>" and "<old-python-version>" should be chosen to match the existing "benchmark.out" files from other tests.
```
Double check this renders since HTML style `<>` can be interpreted by Markdown. You *may* need to escape them.

krefson replied:
I have changed the quotes so this renders correctly.

krefson committed:
Corrected typos and made changes as suggested by Jacob Wilkins.
Added sentence on how to further investigate cause of test failure.
Added description of profiling options.

albapa reviewed (approved):
This is very useful, thanks for collecting/updating the information.
